plugins/plugin-github/che-plugin-github-factory-resolver/src/main/java/org/eclipse/che/plugin/github/factory/resolver/GithubFactoryParametersResolver.java

```diff
@@ -69,27 +83,34 @@ public FactoryDto createFactory(@NotNull final Map<String, String> factoryParame
       throws BadRequestException {
 
     // no need to check null value of url parameter as accept() method has performed the check
-    final GithubUrl githubUrl = githubUrlParser.parse(factoryParameters.get("url"));
+    final GithubUrl githubUrl = githubUrlParser.parse(factoryParameters.get(URL_PARAMETER_NAME));
 
     // create factory from the following location if location exists, else create default factory
-    FactoryDto factory = urlFactoryBuilder.createFactory(githubUrl.factoryJsonFileLocation());
-
+    FactoryDto factory =
+        urlFactoryBuilder.createFactoryFromDevfile(githubUrl.devfileFileLocation());
+    if (factory == null) {
+      factory = urlFactoryBuilder.createFactoryFromJson(githubUrl.factoryJsonFileLocation());
+    }
+    // try to build factory from parameters
+    if (factory == null) {
+      factory = newDto(FactoryDto.class).withV(CURRENT_VERSION);
+    }
     // add workspace configuration if not defined
     if (factory.getWorkspace() == null) {
       factory.setWorkspace(
-          urlFactoryBuilder.buildWorkspaceConfig(
-              githubUrl.getRepository(), githubUrl.getUsername(), githubUrl.dockerFileLocation()));
+          urlFactoryBuilder.buildDefaultWorkspaceConfig(githubUrl.getRepository()));
     }
 
-    // Compute project configuration
-    ProjectConfigDto projectConfigDto =
-        newDto(ProjectConfigDto.class)
-            .withSource(githubSourceStorageBuilder.build(githubUrl))
-            .withName(githubUrl.getRepository())
-            .withType("blank")
-            .withPath("/".concat(githubUrl.getRepository()));
-
-    // apply merging operation from existing and computed settings
-    return projectConfigDtoMerger.merge(factory, projectConfigDto);
+    // apply merging operation from existing and computed settings if needed
+    return projectConfigDtoMerger.merge(
+        factory,
+        () -> {
+          // Compute project configuration
+          return newDto(ProjectConfigDto.class)
+              .withSource(githubSourceStorageBuilder.build(githubUrl))
+              .withName(githubUrl.getRepository())
+              .withType("blank")
```

Comment:
Type field is not needed anymore according to Sun's description https://github.com/eclipse/che/issues/12217.
I think it would be better to leave it empty and make GWT IDE asking project type instead of hardcoding it. WDYT?

Reply:
Removed `blank` here and made as optional factory parameter